plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/SigV4AwsProperties.java

```diff
@@ -17,36 +17,111 @@
+import static org.apache.iceberg.aws.AwsProperties.REST_SIGNER_REGION;
+import static org.apache.iceberg.aws.AwsProperties.REST_SIGNING_NAME;
 
 public class SigV4AwsProperties
         implements AwsProperties
 {
+    private final Optional<StsAssumeRoleCredentialsProvider> iamRoleCredentialsProvider;
     private final Map<String, String> properties;
 
+    private Optional<AwsSessionCredentials> currentAwsSessionCredentials = Optional.empty();
+    private boolean hasIamRoleSessionCredentialsRefreshed;
+
     @Inject
     public SigV4AwsProperties(IcebergSecurityConfig securityConfig, IcebergRestCatalogSigV4Config sigV4Config, S3FileSystemConfig s3Config)
     {
         // TODO https://github.com/trinodb/trino/issues/24916 Allow write operations with SigV4
         checkArgument(securityConfig.getSecuritySystem() == READ_ONLY, "Read-only security system is required");
-        this.properties = ImmutableMap.<String, String>builder()
-                .put("rest.sigv4-enabled", "true")
-                .put("rest.signing-name", sigV4Config.getSigningName())
-                .put("rest.access-key-id", requireNonNull(s3Config.getAwsAccessKey(), "s3.aws-access-key is null"))
-                .put("rest.secret-access-key", requireNonNull(s3Config.getAwsSecretKey(), "s3.aws-secret-key is null"))
-                .put("rest.signing-region", requireNonNull(s3Config.getRegion(), "s3.region is null"))
-                .put("rest-metrics-reporting-enabled", "false")
-                .buildOrThrow();
+
+        properties = new HashMap<>();
+        properties.put("rest.sigv4-enabled", "true");
+        properties.put(REST_SIGNING_NAME, sigV4Config.getSigningName());
+        properties.put(REST_SIGNER_REGION, requireNonNull(s3Config.getRegion(), "s3.region is null"));
+        properties.put("rest-metrics-reporting-enabled", "false");
+
+        if (s3Config.getIamRole() != null) {
+            Optional<AwsCredentialsProvider> staticCredentialsProvider = createStaticCredentialsProvider(s3Config);
+            iamRoleCredentialsProvider = Optional.of(
+                    StsAssumeRoleCredentialsProvider.builder()
+                            .refreshRequest(request -> request
+                                    .roleArn(s3Config.getIamRole())
+                                    .roleSessionName("trino-iceberg-rest-catalog")
+                                    .externalId(s3Config.getExternalId()))
+                            .stsClient(createStsClient(s3Config, staticCredentialsProvider))
+                            .asyncCredentialUpdateEnabled(true)
+                            .build());
+        }
+        else {
+            iamRoleCredentialsProvider = Optional.empty();
+            properties.put(REST_ACCESS_KEY_ID, requireNonNull(s3Config.getAwsAccessKey(), "s3.aws-access-key is null"));
+            properties.put(REST_SECRET_ACCESS_KEY, requireNonNull(s3Config.getAwsSecretKey(), "s3.aws-secret-key is null"));
+        }
```

Comment:
Instead of manually resolving here - Can we use additional properties via `AwsProperties` - `client.assume-role.arn` and set them in this case Iceberg library might update them on demand we don't want to resolve them by ourselves.

Reply:
The property didn't work as far as I tested a few days ago. 

@mayankvadariya Could you double-check that?